packages/swapper/src/swappers/RelaySwapper/utils/getTrade.ts

```diff
@@ -0,0 +1,365 @@
+
+    const appFeesBaseUnit = (() => {
+      // Handle app fees based on cross-chain status
+      if (
+        quote.fees?.app?.amount &&
+        quote.fees?.app?.currency &&
+        isValidRelayToken(quote.fees?.app?.currency)
+      ) {
+        const appFeesAsset = quote.fees?.app?.currency
+          ? relayTokenToAsset(quote.fees.app.currency, deps.assetsById)
+          : undefined
+
+        // @TODO: we might need to change this logic when solana and BTC are supported
+        const isNativeCurrencyInput =
+          isNativeEvmAsset(sellAsset.assetId) && sellAsset.chainId === appFeesAsset?.chainId
+
+        // For cross-chain: always add back app fees
+        // For same-chain: only add back if input is native currency
+        if (isCrossChain || isNativeCurrencyInput) {
+          return quote.fees.app.amount
+        }
+      }
+
+      // cross-chain or same-chain with native currency as input are not applicable
+      return '0'
+    })()
+
+    const relayerFeesBuyAssetBaseUnit = (() => {
+      const relayerFeeRelayToken = quote.fees?.relayer?.currency
+      if (relayerFeeRelayToken && isValidRelayToken(relayerFeeRelayToken)) {
+        const relayerFeesAsset = relayTokenToAsset(relayerFeeRelayToken, deps.assetsById)
+        const relayerFeeAmount = quote.fees?.relayer?.amount ?? '0'
+
+        // If fee is already in buy asset, return as is
+        if (relayerFeesAsset?.assetId === buyAsset.assetId) {
+          return relayerFeeAmount
+        }
+
+        // if fee is in sell asset, convert to buy asset
+        if (relayerFeesAsset?.assetId === sellAsset.assetId) {
+          return convertPrecision({
+            value: relayerFeeAmount,
+            inputExponent: relayerFeesAsset.precision,
+            outputExponent: buyAsset.precision,
+          })
+            .times(rate)
+            .toFixed(0)
+        }
+
+        // If fee is in a different asset, convert to buy asset
+        const feeAmountUsd = quote.fees?.relayer?.amountUsd ?? '0'
+        const buyAssetUsd = quote.details?.currencyOut?.amountUsd ?? '0'
+        const buyAssetAmountBaseUnit = quote.details?.currencyOut?.amount ?? '1'
+
+        if (feeAmountUsd && buyAssetUsd && buyAssetAmountBaseUnit) {
+          // Calculate the rate: (buyAssetAmount / buyAssetUsd) gives us "buy asset per USD"
+          // Then multiply by feeAmountUsd to get the equivalent buy asset amount
+          const buyAssetCryptoBaseUnitPerUsd = bnOrZero(buyAssetAmountBaseUnit).div(buyAssetUsd)
+          const buyAssetFeesCryptoBaseUnit = bnOrZero(feeAmountUsd).times(
+            buyAssetCryptoBaseUnitPerUsd,
+          )
+
+          return buyAssetFeesCryptoBaseUnit.toFixed(0)
+        }
+
+        return '0'
+      }
+
+      return '0'
+    })()
+
+    // Add back relayer service and gas fees (relayer is including both) since they are downsides
+    // And add appFees
+    const buyAmountBeforeFeesCryptoBaseUnit = bnOrZero(currencyOut.minimumAmount)
```

Comment:
I know most of the time we use the minimum amount from the quote, for relay this is very low and make it the latest swapper most of the case, but that's a lie because the end amount is always higher than this one

We might want to make relay a special case and use `currencyOut.amount` instead

Reply:
I'm not sure I understand, that *is* the minimum amount?

Reply:
We can't do that as that would be lying to users and voiding minimum guarantees, see DM with Relay:
> amountOut is the expected amount , minimumAmountOut is the minimum out we guarantee to return (worst amount after slippage is returned) - both include the fees

That is the purpose of `buyAmountBeforeFeesCryptoBaseUnit` - take the final amount (as in, the one that is guaranteed) and add back fees on top of that. Same thing for `buyAmountAfterFeesCryptoBaseUnit`, it should be a close representation of the minimum guaranteed amount.

Reply:
Alright, we already use minimumAmount anyway, except on relayer fee maths that I changed in [29fc44d](https://github.com/shapeshift/web/pull/9188/commits/29fc44d8ce158086f2cbec2c3c7ee0dd44302dac)